pysatSeasons/avg.py

```diff
@@ -1,304 +1,345 @@
+
+                        # For each possible y index
                         for yj in yarr:
-                            # select data with this y index (and we already
+                            # Select data with this y index (and we already
                             # filtered for this x index)
                             yindex, = np.where(yind == yj)
+
                             if len(yindex) > 0:
-                                # for each data product label zk
+                                # For each data product label zk
                                 for zk in zarr:
-                                    # take the data (already filtered by x)
-                                    # filter it by y and
-                                    # select the data product, put it in a
-                                    # list, and extend the deque
-                                    indlab = yData.columns.get_loc(data_label[zk])
-                                    ans[zk][yj][xi].extend(yData.iloc[yindex,
-                                                                      indlab].tolist())
+                                    # Take the data (already filtered by x),
+                                    # filter it by y, select the data product,
+                                    # put it in a list, and extend the deque.
+                                    ans[zk][yj][xi].extend(
+                                        yinst[yindex, data_label[zk]])
 
     return _calc_2d_median(ans, data_label, binx, biny, xarr, yarr, zarr,
-                           numx, numy, numz, returnData)
+                           numx, numy, numz, returnData, return_data)
 
 
 def _calc_2d_median(ans, data_label, binx, biny, xarr, yarr, zarr, numx,
-                    numy, numz, returnData=False):
+                    numy, numz, returnData=None, return_data=False):
+    """Return a 2D average of nD `data_label` over season and `label1` `label2`.
+
+    Parameters
+    ----------
+    ans : list of lists
+        List of lists containing binned data. Provided by `median2D`.
+    data_label : str
+        Label for data to be binned and averaged.
+    binx, biny : array-like
+        List holding [min, max, number of bins] or array-like containing
+        bin edges, where * = 1, 2.
+    xarr, yarr, zarr : list-like
+        Indexing array along bin directions x, y, and data dimension z.
+    numx, numy, numz : int
+        Number of elements along xarr, yarr, zarr.
+    returnData : bool or NoneType
+        If True, also return binned data used to calculate the average in
+        the output dictionary as 'data', in addition to the statistical outputs.
+        Deprecated in favor of `return_data`.
+        (default=None)
+    return_data : bool
+        If True, also return binned data used to calculate the average in
+        the output dictionary as 'data', in addition to the statistical outputs.
+        (default=False)
```

Comment:
args not kwargs

Reply:
The `return_data` variants are both kwargs.